src/functions-reference/higher-order_functions.Rmd

```diff
@@ -141,6 +142,131 @@ Stan's automatic differentiation.
+Solves the ODE system for the times provided using the backward
+differentiation formula (BDF) method with the implementation from
+CVODES with additional control parameters for the CVODES solver.
+
+### Arguments to the ODE solvers
+
+The arguments to the ODE solvers in both the stiff and non-stiff cases
+are as follows.
+
+*   *`ode`*: ODE system function with signature `(real, vector, ...):vector`
+
+The arguments represent (1) time, (2) system state, and a sequence of arguments of any type passed unmodified from the integration function call. The return value contains the derivatives with respect to time of the state.
+
+*   *`initial_state`*: initial state, type `vector`,
+
+*   *`initial_time`*: initial time, type `int` or `real`,
+
+*   *`times`*: solution times, type `real[]`,
+
+*   *`...`*: arguments that will be passed through unmodified to the ODE system function. The types here must match the types in the `...` portion of the ode system function arguments.
+
+For the versions of the ode integrator functions ending in `_tol`, these three
+parameters must also be provided after `times` and before the `...` arguments:
+
+*   `data`   *`rel_tol`*: relative tolerance for the ODE  solver, type `real`,
+data only,
+
+*   `data`   *`abs_tol`*: absolute tolerance for the ODE  solver, type `real`,
+data only, and
+
+*   `data`   *`max_num_steps`*: maximum number of steps to take between output
+times in the ODE solver, type `int`, data only.
+
+### Specifying the ODE System Function
+
+The ODE system function must have the a `vector` return type, and the first two arguments must be a `real` and `vector` in that order. These two arguments can be followed by any number of other arguments of any type:
+
+```
+ vector ode(real time, vector state, ...)
+```
+
+`...` are any number of arguments passed through unmodified from the ODE integrate
+call.
+
+The ODE system function should return the derivative of the state with
+respect to time at the time and state provided. The length of the returned
+vector must match the length of the state input into the function.
+
+The arguments to this function are:
+
+*   *`time`*, the time to evaluate the ODE system
+
+*   *`state`*, the state of the ODE system at the time specified
+
+*   *...*, all other arguments passed through from the integrate call. Each
+argument can be any type.
+
+#### Return values
+
+The return value for the ODE solvers is an array of vectors (type `vector[]`),
+with values consisting of solutions at the specified times.
+
+#### Sizes and parallel arrays
+
+The sizes must match, and in particular, the following groups are of
+the same size:
+
+*   state variables passed into the system function, derivatives
+returned by the system function, initial state passed into the
+solver, and length of each vector in the output,
+
+*   solution times and number of vectors in the output,
+
+## Old integrate_ode_X ODE Interface {#old-functions-ode-solver}
```

Comment:
Maybe call this section "Deprecated `integrate_ode_*` ODE Interface" ?

Reply:
It's probably best to start an appendix with deprecated functions as we do in the reference manual.